src/codesign.py

```diff
@@ -189,13 +209,61 @@ def inverse_pass(self):
         self.symbolic_sim.calculate_edp(self.hw)
         self.symbolic_sim.save_edp_to_file()
 
-        self.inverse_edp = self.symbolic_sim.edp.subs(self.tech_params)
+        # print(f"tech params: {self.tech_params}")
+        # print(f"type of tech_param keys: {type(list(self.tech_params.keys())[0])}")
+
+        # for symbol in self.symbolic_sim.edp.free_symbols:
+        #     for key in self.tech_params.keys():
+        #         # print(f"key: {key}")
+        #         # print(f"symbol: {symbol}")
+        #         if key.name == symbol.name:
+        #             print(f"key: {key} == symbol: {symbol}")
+        #             if key != symbol:
+        #                 print(f"{key}: {key.assumptions0}")
+        #                 print(f"{symbol}: {symbol.assumptions0}")
+        #         # else:
+                #     print(f'key: {key} != symbol: {symbol}')
+            # assert symbol in self.tech_params.keys()
+
+        # tech_params2 = {}
+        # for key in self.tech_params.keys():
+        #     tech_params2[key.name] = self.tech_params[key]
+
+        # print(f"tech params 2: {tech_params2}")
```

Comment:
this might have been from my branch that you merged. but let's remove all this now.

Reply:
^ remove all these prints